.github/workflows/code-freeze.yml

```diff
@@ -0,0 +1,35 @@
+name: Code Freeze
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  FROZEN: "false"  # Set to "true" to freeze branches, "false" otherwise
```

Comment:
Does this make use of action variables? https://github.com/amazon-braket/amazon-braket-sdk-python/settings/variables/actions

Reply:
No these are just set within the workflow. Is there a benefit to moving to action variables until we want to reuse these values?

Reply:
I don't mind the workflow for updating action variables, but there's a limitation that you can't set an empty value (for allowing all feature branch merges) and there's no documentation to attach to the place they get updated.

We could update this workflow to use a more generable/intuitive env variable `UNFROZEN_PREFIX` to allow us to specify `feature` or `feature/my-feature` and not rely on an empty value

Reply:
Updated to use repository variables

Reply:
For code freezes, will we need to publish PRs each time? If so, I think using the action variables would allow for easier changes and cleaner commit history.

Reply:
Makes sense, I've update the script to use action variables